tests/format/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -626,6 +626,75 @@ function enumerable(value) {
 ================================================================================
 `;
 
+exports[`destructuring-private.js [acorn] format 1`] = `
+"Unexpected token (5:13)
+  3 |   constructor() {
+  4 |     console.log(this.#x); // => 1
+> 5 |     const { #x: x } = this;
+    |             ^
+  6 |     console.log(x); // => 1
+  7 |   }
+  8 |   equals({ #x: otherX }) {"
+`;
+
+exports[`destructuring-private.js [espree] format 1`] = `
+"Unexpected token #x (5:13)
+  3 |   constructor() {
+  4 |     console.log(this.#x); // => 1
+> 5 |     const { #x: x } = this;
+    |             ^
+  6 |     console.log(x); // => 1
+  7 |   }
+  8 |   equals({ #x: otherX }) {"
+`;
+
+exports[`destructuring-private.js [meriyah] format 1`] = `
+"[5:13]: Unexpected token: 'PrivateField' (5:13)
+  3 |   constructor() {
+  4 |     console.log(this.#x); // => 1
+> 5 |     const { #x: x } = this;
+    |             ^
+  6 |     console.log(x); // => 1
+  7 |   }
+  8 |   equals({ #x: otherX }) {"
+`;
+
+exports[`destructuring-private.js format 1`] = `
+====================================options=====================================
+parsers: ["babel", "babel-ts", "babel-flow"]
+printWidth: 80
+                                                                                | printWidth
+=====================================input======================================
+class Foo {
+  #x = 1;
+  constructor() {
+    console.log(this.#x); // => 1
+    const { #x: x } = this;
+    console.log(x); // => 1
+  }
+  equals({ #x: otherX }) {
+    const { #x: currentX } = this;
+    return currentX === otherX;
+  }
+}
+
+=====================================output=====================================
+class Foo {
+  #x = 1;
+  constructor() {
+    console.log(this.#x); // => 1
+    const { #x: x } = this;
+    console.log(x); // => 1
+  }
+  equals({ #x: otherX }) {
+    const { #x: currentX } = this;
```

Comment:
Is assign allowed in this syntax?

Reply:
Yes. This example is copied from the example of https://github.com/tc39/proposal-destructuring-private.

Reply:
```js
({ #x: variable.or.property.to.assign } = this);
```

Reply:
Ah sorry I misunderstood. Destructuring private in assign expression is maybe invalid.

Reply:
@fisker Maybe this is valid. So what should we do for it?

Reply:
Seems working fine, 

**Prettier pr-12276**
[Playground link](https://deploy-preview-12276--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAucAbAhgZ0wAgII7AA6UOOJZAtnDABYQAmAFAJSEVk5PA4DEAHkhwA3dACcAlugBGqOADoIY+QAcxEFXDEwAnvJgR5WTBIDmpAL44AvDjoTMLANwcyFkhZAAaEBpgToTGRQcXUAdwAFcQQglHRUMPQdIJ9pMXQwAGsaAGV0agAZCSg4ZAAzeMw4VPSs3JUM4tNkGDEAV2qQOEppOAYGPoL0KFM29FM4ADElSnQYfxHkEHQ2g28QWhhKVAB1Wgl4TAawOByYg4lhA50lsGx14qrtCPTTWfLKzoArTH4cprkAEU2hB4B9UFUfA0xE8ltIZHBUOs1MUYDsJAw6MgABwABih6iqO3SKiWajgT2EpR8AEcQfAXhpYstMABaEp9PrrMRwOkSHkvcbvJAVCGdKqUCQtdrigFwYGg0oiz4+GAydGY2jIABMqvSElQTQAwhBKMKupgAKzrNpVAAqMlioshIGEHQAklABrAcmBJCoYHgvTldHJwVULBYgA)
<!-- prettier-ignore -->
```sh
--parser babel
```

**Input:**
<!-- prettier-ignore -->
```jsx
class A {
  
  method() {
    ({ #x: variable.or.property.to.assign } = this);
    }
}
```

**Output:**
<!-- prettier-ignore -->
```jsx
class A {
  method() {
    ({ #x: variable.or.property.to.assign } = this);
  }
}

```

Add a test for it, and check if we need adjust the printer.

This syntax is a good one, I was wait for it for quite a while, I may actual use it.

Reply:
Yes, valid. https://github.com/tc39/proposal-destructuring-private/issues/4#issue-1045243014

Reply:
Don't forget add some tests for destructuring with default value.